packages/rest/src/rest.component.ts

```diff
@@ -37,6 +44,26 @@ export class RestComponent implements Component {
     [RestBindings.SequenceActions.PARSE_PARAMS.key]: ParseParamsProvider,
     [RestBindings.SequenceActions.SEND.key]: SendProvider,
   };
+  /**
+   * Add built-in body parsers
+   */
+  bindings = [
+    Binding.bind(RestBindings.REQUEST_BODY_PARSER)
+      .toClass(RequestBodyParser)
+      .inScope(BindingScope.SINGLETON),
+    Binding.bind(RestBindings.REQUEST_BODY_PARSER_JSON)
+      .toClass(JsonBodyParser)
```

Comment:
+1 for using the new `bindings` API.

I wish we could reuse the implementation from `RestServer#bodyParser` here. Can we perhaps at least re-implement `RestServer#bodyParser` as a private helper method in the component class (just copy&paste the implementation details) and simplify this long registration block as follows?

```ts
export class RestComponent implements Component {
  // ...
  bindings: [];

  constructor(/*...*/) {
    // ...

    this.bodyParser(JsonBodyParser);
    this.bodyParser(TextBodyParser);
    this.bodyParser(UrlEncodedBodyParser);
  }
}
```


Also, how do we expect 3rd-party extensions to contribute custom body parsers? I think we should provide them with a helper that will construct a binding with the correct tag, scope and inferred key for them. To keep this pull request simple, I am proposing to write a new helper function that will accept a parser constructor and return a Binding. This function should be used by `RestServer#bodyParser`. As for `RestComponent`, we can provide a helper method `bodyParser` calling this helper function under the hood, or simply call the helper to create Binding instances to initialize `bindings` array. For example:

```ts
export class RestComponent implements Component {
  // ...
  bindings: [
    Binding.bind(RestBindings.REQUEST_BODY_PARSER)
       .toClass(RequestBodyParser)
       .inScope(BindingScope.SINGLETON),
    createBodyParserBinding(JsonBodyParser),
    createBodyParserBinding(TextBodyParser),
    createBodyParserBinding(UrlEncodedBodyParser),
  ];

  // ...
}
```

I think I like this last approach most, because it's using the same constructs that any 3rd-party extension author can use too.

We need to start writing a guide for extension developers to show them how to contribute custom body parsers, the guide should mention this new helper function for creating bindings too.